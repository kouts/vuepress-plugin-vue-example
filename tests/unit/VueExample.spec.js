import { mount, shallowMount } from '@vue/test-utils';
import VueExample from '@/VueExample';
jest.mock('@dynamic/loadComponent');
import { loadComponent, loadComponentAsString } from '@dynamic/loadComponent';

const countLines = (str) => str.split(/\r\n|\r|\n/).length;

const props = {
  file: 'test-component.vue'
};

let wrapper;
let contents;

beforeEach(async() => {
  wrapper = shallowMount(VueExample, {
    propsData: props
  });
  contents = await loadComponentAsString();
});

afterEach(() => {
  wrapper.destroy();
});

describe('VueExample', () => {

  it('renders correctly', () => {
    const div = wrapper.find(`div.card`);
    // console.log(wrapper.html());
  });

  it('parses the template SFC sections', async () => {
    const parsed = wrapper.vm.parseSfcSection('template', contents);
    expect(countLines(parsed)).toBe(7);
    expect(parsed).toContain('<template>');
    expect(parsed).toContain('</template>');
    expect(parsed).not.toContain('<script>');
    expect(parsed).not.toContain('</script>');
    expect(parsed).not.toContain('<style');
    expect(parsed).not.toContain('</style>');
  });

  it('parses the script SFC sections', async () => {
    const parsed = wrapper.vm.parseSfcSection('script', contents);
    expect(countLines(parsed)).toBe(16);
    expect(parsed).toContain('<script>');
    expect(parsed).toContain('</script>');
    expect(parsed).not.toContain('<template>');
    expect(parsed).not.toContain('</template>');
    expect(parsed).not.toContain('<style');
    expect(parsed).not.toContain('</style>');
  });

  it('parses the style SFC sections', async () => {
    const parsed = wrapper.vm.parseSfcSection('style', contents);
    expect(countLines(parsed)).toBe(13);
    expect(parsed).toContain('<style');
    expect(parsed).toContain('</style>');
    expect(parsed).not.toContain('<template>');
    expect(parsed).not.toContain('</template>');
    expect(parsed).not.toContain('<script>');
    expect(parsed).not.toContain('</script>');
  });

  it('removes comments from the template section', async () => {
    const str  = `
    <template>
      <!-- This is a comment -->
      <p>Text</p>
    </template>
    `;
    const removed = wrapper.vm.removeComments('template', str);
    expect(removed).not.toContain('<!-- This is a comment -->');
  });

  it('removes comments from the script section', async () => {
    const str  = `
    <script>
      // This is a comment
      const test = () => 'test';
    </script>
    `;
    const removed = wrapper.vm.removeComments('script', str);
    expect(removed).not.toContain('// This is a comment');
  });

  it('removes comments from the style section', async () => {
    const str  = `
    <style>
      /* This is a comment */
      .btn-primary {
        display: inline-block;
      }
    </style>
    `;
    const removed = wrapper.vm.removeComments('script', str);
    expect(removed).not.toContain('// This is a comment');
  });    

});
