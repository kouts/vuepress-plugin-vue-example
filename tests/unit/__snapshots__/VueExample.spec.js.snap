// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VueExample > renders correctly 1`] = `
"<div data-v-54d51888="" class="card">
  <div data-v-54d51888="" class="card-header d-flex bg-white justify-content-between">
    <ul data-v-54d51888="" class="nav nav-pills">
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link active">Example</a></li>
    </ul>
    <ul data-v-54d51888="" class="nav nav-pills justify-content-end">
      <!--v-if-->
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link"><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="2" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path data-v-54d51888="" stroke="none" d="M0 0h24v24H0z" fill="none"></path>
            <polyline data-v-54d51888="" points="7 8 3 12 7 16"></polyline>
            <polyline data-v-54d51888="" points="17 8 21 12 17 16"></polyline>
            <line data-v-54d51888="" x1="14" y1="4" x2="10" y2="20"></line>
          </svg>
          <!--v-if-->
          <!--v-if-->Template
        </a></li>
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link">
          <!--v-if--><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="1.5" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path stroke="none" d="M0 0h24v24H0z" fill="none" data-v-54d51888=""></path>
            <path d="M14 3v4a1 1 0 0 0 1 1h4" data-v-54d51888=""></path>
            <path d="M17 21h-10a2 2 0 0 1 -2 -2v-14a2 2 0 0 1 2 -2h7l5 5v11a2 2 0 0 1 -2 2z" data-v-54d51888=""></path>
            <line x1="9" y1="9" x2="10" y2="9" data-v-54d51888=""></line>
            <line x1="9" y1="13" x2="15" y2="13" data-v-54d51888=""></line>
            <line x1="9" y1="17" x2="15" y2="17" data-v-54d51888=""></line>
          </svg>
          <!--v-if-->Script
        </a></li>
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link">
          <!--v-if-->
          <!--v-if--><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="2" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path stroke="none" d="M0 0h24v24H0z" fill="none" data-v-54d51888=""></path>
            <path d="M14 3v4a1 1 0 0 0 1 1h4" data-v-54d51888=""></path>
            <path d="M17 21h-10a2 2 0 0 1 -2 -2v-14a2 2 0 0 1 2 -2h7l5 5v11a2 2 0 0 1 -2 2z" data-v-54d51888=""></path>
            <path d="M10 13l-1 2l1 2" data-v-54d51888=""></path>
            <path d="M14 13l1 2l-1 2" data-v-54d51888=""></path>
          </svg>Style
        </a></li>
      <li data-v-54d51888="" class="d-flex align-items-center"><a data-v-54d51888="" href="#" class="ml-2 mr-n2 expand-collapse leading-none"><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" stroke-width="1.5" stroke="#000000" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path data-v-54d51888="" stroke="none" d="M0 0h24v24H0z" fill="none"></path>
            <path data-v-54d51888="" d="M9 10l3 -3l3 3"></path>
            <path data-v-54d51888="" d="M9 14l3 3l3 -3"></path>
          </svg></a></li>
    </ul>
  </div>
  <div data-v-54d51888="" class="card-body">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <!---->
  </div>
</div>"
`;

exports[`VueExample > renders the script section 1`] = `
"<div data-v-54d51888="" class="card">
  <div data-v-54d51888="" class="card-header d-flex bg-white justify-content-between">
    <ul data-v-54d51888="" class="nav nav-pills">
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link">Example</a></li>
    </ul>
    <ul data-v-54d51888="" class="nav nav-pills justify-content-end">
      <!--v-if-->
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link"><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="2" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path data-v-54d51888="" stroke="none" d="M0 0h24v24H0z" fill="none"></path>
            <polyline data-v-54d51888="" points="7 8 3 12 7 16"></polyline>
            <polyline data-v-54d51888="" points="17 8 21 12 17 16"></polyline>
            <line data-v-54d51888="" x1="14" y1="4" x2="10" y2="20"></line>
          </svg>
          <!--v-if-->
          <!--v-if-->Template
        </a></li>
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link active">
          <!--v-if--><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="1.5" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path stroke="none" d="M0 0h24v24H0z" fill="none" data-v-54d51888=""></path>
            <path d="M14 3v4a1 1 0 0 0 1 1h4" data-v-54d51888=""></path>
            <path d="M17 21h-10a2 2 0 0 1 -2 -2v-14a2 2 0 0 1 2 -2h7l5 5v11a2 2 0 0 1 -2 2z" data-v-54d51888=""></path>
            <line x1="9" y1="9" x2="10" y2="9" data-v-54d51888=""></line>
            <line x1="9" y1="13" x2="15" y2="13" data-v-54d51888=""></line>
            <line x1="9" y1="17" x2="15" y2="17" data-v-54d51888=""></line>
          </svg>
          <!--v-if-->Script
        </a></li>
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link">
          <!--v-if-->
          <!--v-if--><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="2" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path stroke="none" d="M0 0h24v24H0z" fill="none" data-v-54d51888=""></path>
            <path d="M14 3v4a1 1 0 0 0 1 1h4" data-v-54d51888=""></path>
            <path d="M17 21h-10a2 2 0 0 1 -2 -2v-14a2 2 0 0 1 2 -2h7l5 5v11a2 2 0 0 1 -2 2z" data-v-54d51888=""></path>
            <path d="M10 13l-1 2l1 2" data-v-54d51888=""></path>
            <path d="M14 13l1 2l-1 2" data-v-54d51888=""></path>
          </svg>Style
        </a></li>
      <li data-v-54d51888="" class="d-flex align-items-center"><a data-v-54d51888="" href="#" class="ml-2 mr-n2 expand-collapse leading-none"><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" stroke-width="1.5" stroke="#000000" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path data-v-54d51888="" stroke="none" d="M0 0h24v24H0z" fill="none"></path>
            <path data-v-54d51888="" d="M9 10l3 -3l3 3"></path>
            <path data-v-54d51888="" d="M9 14l3 3l3 -3"></path>
          </svg></a></li>
    </ul>
  </div>
  <div data-v-54d51888="" class="card-body">
    <!--v-if-->
    <!--v-if-->
    <div data-v-54d51888="">
      <div data-v-e6ee2242="" data-v-54d51888="" class="language-javascript"><pre data-v-e6ee2242="" class="language-javascript"><code><span class="token operator">&lt;</span>script<span class="token operator">&gt;</span>
<span class="token keyword">export</span> <span class="token keyword">default</span> <span class="token punctuation">{</span>
  <span class="token literal-property property">name</span><span class="token operator">:</span> <span class="token string">'TestComponentText'</span><span class="token punctuation">,</span>
  
  <span class="token function">data</span><span class="token punctuation">(</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>
    <span class="token keyword">return</span> <span class="token punctuation">{</span>
      <span class="token literal-property property">count</span><span class="token operator">:</span> <span class="token number">0</span><span class="token punctuation">,</span>
    <span class="token punctuation">}</span>
  <span class="token punctuation">}</span><span class="token punctuation">,</span>
  <span class="token literal-property property">methods</span><span class="token operator">:</span> <span class="token punctuation">{</span>
    <span class="token function">handleClick</span><span class="token punctuation">(</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>
      <span class="token keyword">this</span><span class="token punctuation">.</span>count<span class="token operator">++</span>
      console<span class="token punctuation">.</span><span class="token function">log</span><span class="token punctuation">(</span><span class="token string">'clicked'</span><span class="token punctuation">,</span> <span class="token keyword">this</span><span class="token punctuation">.</span>count<span class="token punctuation">)</span>
    <span class="token punctuation">}</span><span class="token punctuation">,</span>
  <span class="token punctuation">}</span><span class="token punctuation">,</span>
<span class="token punctuation">}</span>
<span class="token operator">&lt;</span><span class="token operator">/</span>script<span class="token operator">&gt;</span></code></pre>
      </div>
    </div>
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;

exports[`VueExample > renders the style section 1`] = `
"<div data-v-54d51888="" class="card">
  <div data-v-54d51888="" class="card-header d-flex bg-white justify-content-between">
    <ul data-v-54d51888="" class="nav nav-pills">
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link">Example</a></li>
    </ul>
    <ul data-v-54d51888="" class="nav nav-pills justify-content-end">
      <!--v-if-->
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link"><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="2" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path data-v-54d51888="" stroke="none" d="M0 0h24v24H0z" fill="none"></path>
            <polyline data-v-54d51888="" points="7 8 3 12 7 16"></polyline>
            <polyline data-v-54d51888="" points="17 8 21 12 17 16"></polyline>
            <line data-v-54d51888="" x1="14" y1="4" x2="10" y2="20"></line>
          </svg>
          <!--v-if-->
          <!--v-if-->Template
        </a></li>
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link">
          <!--v-if--><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="1.5" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path stroke="none" d="M0 0h24v24H0z" fill="none" data-v-54d51888=""></path>
            <path d="M14 3v4a1 1 0 0 0 1 1h4" data-v-54d51888=""></path>
            <path d="M17 21h-10a2 2 0 0 1 -2 -2v-14a2 2 0 0 1 2 -2h7l5 5v11a2 2 0 0 1 -2 2z" data-v-54d51888=""></path>
            <line x1="9" y1="9" x2="10" y2="9" data-v-54d51888=""></line>
            <line x1="9" y1="13" x2="15" y2="13" data-v-54d51888=""></line>
            <line x1="9" y1="17" x2="15" y2="17" data-v-54d51888=""></line>
          </svg>
          <!--v-if-->Script
        </a></li>
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link active">
          <!--v-if-->
          <!--v-if--><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="2" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path stroke="none" d="M0 0h24v24H0z" fill="none" data-v-54d51888=""></path>
            <path d="M14 3v4a1 1 0 0 0 1 1h4" data-v-54d51888=""></path>
            <path d="M17 21h-10a2 2 0 0 1 -2 -2v-14a2 2 0 0 1 2 -2h7l5 5v11a2 2 0 0 1 -2 2z" data-v-54d51888=""></path>
            <path d="M10 13l-1 2l1 2" data-v-54d51888=""></path>
            <path d="M14 13l1 2l-1 2" data-v-54d51888=""></path>
          </svg>Style
        </a></li>
      <li data-v-54d51888="" class="d-flex align-items-center"><a data-v-54d51888="" href="#" class="ml-2 mr-n2 expand-collapse leading-none"><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" stroke-width="1.5" stroke="#000000" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path data-v-54d51888="" stroke="none" d="M0 0h24v24H0z" fill="none"></path>
            <path data-v-54d51888="" d="M9 10l3 -3l3 3"></path>
            <path data-v-54d51888="" d="M9 14l3 3l3 -3"></path>
          </svg></a></li>
    </ul>
  </div>
  <div data-v-54d51888="" class="card-body">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
    <div data-v-54d51888="">
      <div data-v-e6ee2242="" data-v-54d51888="" class="language-css"><pre data-v-e6ee2242="" class="language-css"><code><span class="token selector">&lt;style scoped&gt;
.btn-primary</span> <span class="token punctuation">{</span>
  <span class="token property">display</span><span class="token punctuation">:</span> inline-block<span class="token punctuation">;</span>
  <span class="token property">font-size</span><span class="token punctuation">:</span> 1.2rem<span class="token punctuation">;</span>
  <span class="token property">color</span><span class="token punctuation">:</span> #fff<span class="token punctuation">;</span>
  <span class="token property">background-color</span><span class="token punctuation">:</span> #3eaf7c<span class="token punctuation">;</span>
  <span class="token property">padding</span><span class="token punctuation">:</span> 0.8rem 1.6rem<span class="token punctuation">;</span>
  <span class="token property">border-radius</span><span class="token punctuation">:</span> 4px<span class="token punctuation">;</span>
  <span class="token property">transition</span><span class="token punctuation">:</span> background-color 0.1s ease<span class="token punctuation">;</span>
  <span class="token property">box-sizing</span><span class="token punctuation">:</span> border-box<span class="token punctuation">;</span>
  <span class="token property">border-bottom</span><span class="token punctuation">:</span> 1px solid #389d70<span class="token punctuation">;</span>
<span class="token punctuation">}</span>
&lt;/style&gt;</code></pre>
      </div>
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`VueExample > renders the template section 1`] = `
"<div data-v-54d51888="" class="card">
  <div data-v-54d51888="" class="card-header d-flex bg-white justify-content-between">
    <ul data-v-54d51888="" class="nav nav-pills">
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link">Example</a></li>
    </ul>
    <ul data-v-54d51888="" class="nav nav-pills justify-content-end">
      <!--v-if-->
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link active"><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="2" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path data-v-54d51888="" stroke="none" d="M0 0h24v24H0z" fill="none"></path>
            <polyline data-v-54d51888="" points="7 8 3 12 7 16"></polyline>
            <polyline data-v-54d51888="" points="17 8 21 12 17 16"></polyline>
            <line data-v-54d51888="" x1="14" y1="4" x2="10" y2="20"></line>
          </svg>
          <!--v-if-->
          <!--v-if-->Template
        </a></li>
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link">
          <!--v-if--><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="1.5" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path stroke="none" d="M0 0h24v24H0z" fill="none" data-v-54d51888=""></path>
            <path d="M14 3v4a1 1 0 0 0 1 1h4" data-v-54d51888=""></path>
            <path d="M17 21h-10a2 2 0 0 1 -2 -2v-14a2 2 0 0 1 2 -2h7l5 5v11a2 2 0 0 1 -2 2z" data-v-54d51888=""></path>
            <line x1="9" y1="9" x2="10" y2="9" data-v-54d51888=""></line>
            <line x1="9" y1="13" x2="15" y2="13" data-v-54d51888=""></line>
            <line x1="9" y1="17" x2="15" y2="17" data-v-54d51888=""></line>
          </svg>
          <!--v-if-->Script
        </a></li>
      <li data-v-54d51888="" class="nav-item"><a data-v-54d51888="" href="#" class="nav-link">
          <!--v-if-->
          <!--v-if--><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" stroke-width="2" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path stroke="none" d="M0 0h24v24H0z" fill="none" data-v-54d51888=""></path>
            <path d="M14 3v4a1 1 0 0 0 1 1h4" data-v-54d51888=""></path>
            <path d="M17 21h-10a2 2 0 0 1 -2 -2v-14a2 2 0 0 1 2 -2h7l5 5v11a2 2 0 0 1 -2 2z" data-v-54d51888=""></path>
            <path d="M10 13l-1 2l1 2" data-v-54d51888=""></path>
            <path d="M14 13l1 2l-1 2" data-v-54d51888=""></path>
          </svg>Style
        </a></li>
      <li data-v-54d51888="" class="d-flex align-items-center"><a data-v-54d51888="" href="#" class="ml-2 mr-n2 expand-collapse leading-none"><svg data-v-54d51888="" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" stroke-width="1.5" stroke="#000000" fill="none" stroke-linecap="round" stroke-linejoin="round">
            <path data-v-54d51888="" stroke="none" d="M0 0h24v24H0z" fill="none"></path>
            <path data-v-54d51888="" d="M9 10l3 -3l3 3"></path>
            <path data-v-54d51888="" d="M9 14l3 3l3 -3"></path>
          </svg></a></li>
    </ul>
  </div>
  <div data-v-54d51888="" class="card-body">
    <!--v-if-->
    <div data-v-54d51888="">
      <div data-v-e6ee2242="" data-v-54d51888="" class="language-markup"><pre data-v-e6ee2242="" class="language-markup"><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>p</span> <span class="token attr-name">class</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>demo<span class="token punctuation">"</span></span><span class="token punctuation">&gt;</span></span>
    <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>button</span> <span class="token attr-name">class</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>btn-primary<span class="token punctuation">"</span></span> <span class="token attr-name">@click.prevent</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>handleClick<span class="token punctuation">"</span></span><span class="token punctuation">&gt;</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>slot</span><span class="token punctuation">&gt;</span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>slot</span><span class="token punctuation">&gt;</span></span>(clicked - {{ count }})<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>button</span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>p</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span></code></pre>
      </div>
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>"
`;
